Common/Data/SubscriptionDataSource.cs

```diff
@@ -44,24 +44,33 @@ public class SubscriptionDataSource : IEquatable<SubscriptionDataSource>
         /// </summary>
         public readonly IReadOnlyList<KeyValuePair<string, string>> Headers;
 
+        /// <summary>
+        /// Flags if algorithm is running on a live server.
+        /// </summary>
+        public readonly bool? IsLiveMode;
+
         /// <summary>
         /// Initializes a new instance of the <see cref="SubscriptionDataSource"/> class.
         /// </summary>
         /// <param name="source">The subscription's data source location</param>
         /// <param name="transportMedium">The transport medium to be used to retrieve the subscription's data from the source</param>
-        public SubscriptionDataSource(string source, SubscriptionTransportMedium transportMedium)
-            : this(source, transportMedium, FileFormat.Csv)
+        /// <param name="isLiveMode">Flags if algorithm is running on a live server.</param>
+        /// <remarks>In case of implementing REST as Transport Medium you must declare if the Algorithm is running in live mode through the <code>isLiveMode</code> argument.</remarks>
+        public SubscriptionDataSource(string source, SubscriptionTransportMedium transportMedium, bool isLiveMode)
```

Comment:
Sadly, changing these constructors would break many many algorithms and would also require users to properly set the value, which is odd considering the `GetSource` method has a `isLiveMode`parameter in it and would just pipe it through. This means that the calling context already has access to this data so it shouldn't be needed on this object.